apps/theming-designer/src/components/FabricPalette.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+import { Card } from '@uifabric/react-cards';
+import { FabricSlots, IThemeRules } from 'office-ui-fabric-react/lib/ThemeGenerator';
+import { Stack } from '../../../../packages/office-ui-fabric-react/lib/index';
+import { mergeStyles } from '@uifabric/merge-styles';
+
+export interface IFabricPaletteProps {
+  themeRules: IThemeRules | undefined;
+}
+
+const tableClassName = mergeStyles({
+  selectors: {
+    th: {
+      padding: 10,
+      textAlign: 'left'
+    },
+    td: {
+      padding: 10,
+      textAlign: 'left'
+    }
+  }
+});
+
+const slotClassName = mergeStyles({
+  display: 'flex',
+  alignItems: 'center',
+  overflow: 'auto'
+});
+
+const fabricPaletteColorBox = mergeStyles({
+  width: 15,
+  height: 15,
+  display: 'inline-block',
+  left: 5,
+  top: 5,
+  border: '1px solid black',
+  flexShrink: 0
+});
+
+export const FabricPalette: React.StatelessComponent<IFabricPaletteProps> = (props: IFabricPaletteProps) => {
+  const fabricSlotWidget = (fabricSlot: FabricSlots): JSX.Element => {
+    const slotRule = props.themeRules![FabricSlots[fabricSlot]];
+    return (
+      <div key={slotRule.name} className={slotClassName}>
+        <Stack horizontal gap={5}>
+          <div
+            key={slotRule.name}
+            className={fabricPaletteColorBox}
+            style={{ backgroundColor: slotRule.color!.str }}
```

Comment:
Having `className` and `style` is kind of redundant, it would be great if you could consolidate both into one.

Reply:
The problem here is I only have access to each of the different fabricSlots inside this function fabricSlotWidget where each different slotRule is calculated based off each of the different fabricSlots passed in. So I can't set it generically inside the fabricPaletteColorBox mergeStyle.